### ISSUE TITLE
Regarding precission and recall

### ISSUE THREAD

**Issue by spartian** [AlexGidiotis/Document-Classifier-LSTM#7]
Hello,

In the hatt_classifier.py file you have defined a function called f1_score. I wanted to separately calculate precision and recall. **While calculating them like in f1_score, do I have to apply tf.reduce_mean function on them?**

**Like for f1_score you have applied the reduce_mean function at the end. Do I have do similar with precision and recall?**

**Comment by AlexGidiotis** [comment]
You can easily implement callbacks for precision and recall (or find some code online). The reduce_mean is required in order to output a scalar mean value.

**Comment by spartian** [comment]
I have already implemented it. But I was asking if we have to mention implement reduce_mean while calculating precision and recall?

**Comment by spartian** [comment]
This means for a scalar score like precision and recall we have to implement this method. Ok got it!!!

Another question .....have you thought of calculating confusion matrix for multi label classification? In sklearn they have now implemented it in the library? I could not find anywhere for tensorflow. What do you think?

**Comment by AlexGidiotis** [comment]
You can use sklearn at test time in order to calculate the confusion matrix. I would not recommend to try and implement it in tensorflow since it is not required for the training.

**Comment by spartian** [comment]
Sorry...as I said earlier that I am beginier in keras.......I didn't understand. Even for test code, I would be using tensorflow objects. So how can i calculate it with sklearn? Do you mean by converting the tensorflow object to numpy array? or am I missing something here?

**Comment by AlexGidiotis** [comment]
You predict with the model and then use the predictions and labels to calculate confusion matrix with sklearn.

**Comment by spartian** [comment]
ok...thank you
